compiler/resolution/functionResolution.cpp

```diff
@@ -3541,6 +3541,73 @@ static void resolveNormalCallConstRef(CallExpr* call) {
   }
 }
 
+static Type* getArrayElementType(AggregateType* arrayType) {
+  Type* instType = arrayType->getField("_instance")->type;
+  AggregateType* instClass = toAggregateType(canonicalClassType(instType));
+  Type* eltType = instClass->getField("eltType")->getValType();
+  return eltType;
+}
+
+static bool isAssignedTo(Symbol* actualSym, SymExpr* knownRef) {
+  for_SymbolUses(use, actualSym)
```

Comment:
The pattern I am looking for is a call to a proc `=`. So `for_SymbolUses` considers it a "use". I added a comment about that.

Reply:
I think it'd be better to use for_SymbolSymExprs here, because the LHS of `=` is not a "use" by any reasonable definition. Nonetheless we don't need to fix that now.